vignettes/compareToBaseR.Rmd

```diff
@@ -0,0 +1,683 @@
+str_dup(fruit, 1:3)
+str_c("ba", str_dup("na", 0:5))
+
+```
+
+## `str_split`
+
+*Split up a string into pieces*
+
+To split a string into pieces with breaks based on a particular pattern match stringr uses `str_split()` and base R uses `strsplit()`. Unlike many of our other examples, when splitting a string into pieces, both base R and stringr use the same default order of inputs (string, then split criteria).
+
+```{r}
+fruits <- c(
+  "apples and oranges and pears and bananas",
+  "pineapples and mangos and guavas"
+)
+
+## stringr
+str_split(fruits, " and ")
+
+## base
+strsplit(fruits, " and ")
+```
+
+The stringr package's `str_split()` allows for more control over the split, including restricting the number of possible matches.
+
+```{r}
+## stringr
+str_split(fruits, " and ", n = 3)
+str_split(fruits, " and ", n = 2)
+
+```
+
+
+
+## `str_glue`
+
+*Format and interpolate a string with glue*
+
+You may want to combine strings and interpolate them at the same time. You can do this with a mix of `paste()` and `eval()` in base R, but stringr makes the process simpler with `str_glue()`.
+
+
+```{r}
+name <- "Fred"
+age <- 50
+anniversary <- as.Date("1991-10-12")
+
+## stringr
+str_glue(
+  "My name is {name}, ",
+  "my age next year is {age + 1}, ",
+  "and my anniversary is {format(anniversary, '%A, %B %d, %Y')}."
+)
+
+## base
+paste("My name is ", eval(name), ", my age next year is ", eval(age + 1), ", and my anniversary is ", eval(format(anniversary, "%A, %B %d, %Y")), ".", sep = "")
+```
+
+```{r}
+## stringr
+mtcars %>% str_glue_data("{rownames(.)} has {hp} hp") %>% head()
+
+## base
+paste(rownames(mtcars), "has", mtcars$hp, "hp") %>% head()
+```
+
+
+# Order strings
+
+## `str_order`
+
+*Order or sort a character vector*
+
+Both base R and stringr have separate functions to order and sort strings.
+
+```{r}
+## stringr
+str_order(letters)
+str_sort(letters)
+
+## base
+order(letters)
+sort(letters)
+```
+
+Some options in `str_order()` and `str_sort()` don't have analogous base R options.
+
+```{r}
+str_order(letters, locale = "haw") ## vowels come first
+str_sort(letters, locale = "haw")
+## no locale capability in order, sort
+```
+
+```{r}
+x <- c("100a10", "100a5", "2b", "2a")
+str_sort(x)
+str_sort(x, numeric = TRUE)
+
+sort(x)
+## no numeric capability in sort
+```
+
+# Helpers
+
+## `str_conv`
+
+*Specify the encoding of a string*
+
+You can specify the encoding of particular strings using `str_conv()` and `iconv()` in stringr and base R respectively.
+
+```{r}
+x <- rawToChar(as.raw(177))
+x
+
+## stringr
+str_conv(x, "ISO-8859-2") # Polish "a with ogonek"
+str_conv(x, "ISO-8859-1") # Plus-minus
+
+## base
+iconv(x, "ISO-8859-2")
+iconv(x, "ISO-8859-1")
+```
+
+
+## `str_view`
+
+*View HTML rendering of regular expression match*
+
+The stringr package provides a nice way to look at regular expression matches to help with text processing. 
+
+```{r}
+## stringr
+str_view(c("abc", "def", "fgh"), "[aeiou]")
+```
+
+
+# Take-Aways
```

Comment:
I think you should move this to before detect matches — that way you can discuss argument order in one place.